source/containers/Form/hooks/formActions.ts

```diff
@@ -330,13 +330,42 @@ export function validateAllStepAnswers( state: FormReducerState, onErrorCallback
     state = validateAnswer(state, { [id]: answer }, id);
   });
 
-  // Find out if any validation failed.
-  const fieldKeys = state.validations;
-  for(const fieldKey in fieldKeys) {
-    if (state.validations[fieldKey]?.isValid === false) {
-      allInputsValid = false;
+  // Find out if any validation failed for types text, input, editableList and repeaterField.
+  // Validation field unique so different handling required for the different types.
+  // Will break at the very first found instance of isValid === false and set allInputsValid = false.
+  for (const questionIndex in currentStepQuestions) {
+    const question = currentStepQuestions[questionIndex]
 
-      break;
+    if (['text', 'number'].includes(question.type)) {             // Checks for isValid in text and number.
```

Comment:
Only because Dan has been sending the CleanCode stuff about unwarranted comments... but this comment feels redundant. No need to repeat what the code says.